### PR TITLE
Add support for Vulkan on Wayland

### PR DESCRIPTION
**Pull request by myfreeweb**
Works on `QT_QPA_PLATFORM=wayland-egl`:

rpcs3-wayland-vulkan

### CODE REVIEW

**Comment by Megamouse** [comment]
please try to adjust to our coding style as described in here: https://github.com/RPCS3/rpcs3/wiki/Coding-Style
only then we can properly review this contribution.

There are a lot of missing or wrong indentations like with the namespaces and you often have opening braces in the same line instead of a new one.

File: rpcs3/rpcs3qt/gs_frame.cpp [hunk 1]
```diff
@@ -145,6 +150,17 @@ void gs_frame::show()
 	});
 }
 
+#ifdef VK_USE_PLATFORM_WAYLAND_KHR
+std::pair<wl_display*, wl_surface*> gs_frame::wayland_handle() const
+{
```
Comment (kd-11) [inline rpcs3/rpcs3qt/gs_frame.cpp]: Move this into gs_frame::handle(). The checks in VKGSRender to check if wayland handle is valid can also be done there.

File: rpcs3/Emu/RSX/VK/VKHelpers.h [hunk 2]
```diff
@@ -1309,16 +1317,36 @@ namespace vk
...
 #elif HAVE_VULKAN
-		
-		vk::swap_chain* createSwapChain(Display *display, Window window, vk::physical_device &dev)
+
+		typedef std::variant<
+			std::pair<Display*, Window>
+#ifdef VK_USE_PLATFORM_WAYLAND_KHR
+			, std::pair<wl_display*, wl_surface*>
+#endif
+		> VkUnixContext;
```
Comment (kd-11) [inline rpcs3/Emu/RSX/VK/VKHelpers.h]: This template can be moved to gsframe header instead as it is not restricted to use with vulkan only and holds general unix window and display properties. Name can be changed to reflect that as well, something like unix_display_context or something along those lines

**Comment by myfreeweb** [comment]
Moved the stuff to gsframe header. New version: `handle()` returns `display_handle_t` which an alias for `HWND` on Windows, and this `variant` on Unix.

also moved some identical code outside of platform ifdefs in VKGSRender.cpp

File: rpcs3/rpcs3qt/gs_frame.cpp [hunk 3]
```diff
@@ -8,6 +8,13 @@
 #include <QTimer>
 #include <QThread>
 
+#ifdef VK_USE_PLATFORM_WAYLAND_KHR
```
Comment (kd-11) [inline rpcs3/rpcs3qt/gs_frame.cpp]: This whole block from line 11 to 16 should be in an ifndef _WIN32 block. That should fix appveyor CI and allow this set to be mergeable

**Comment by JXPZ** [comment]
esboco Drive Geforce 390.65

**Comment by JXPZ** [comment]
1050 TI OC EDITION 4GB GIGABYTE

**Comment by Inviuz** [comment]
@JXPZ Completely unrelated. The issue that talks about problems with newest nvidia driver is already closed. Issue number #4036

**Pull request merged**